pkg/agent/controller/egress/egress_controller.go

```diff
@@ -974,3 +1015,28 @@ func (c *EgressController) GetEgress(ns, podName string) (string, string, error)
 func isEgressSchedulable(egress *crdv1a2.Egress) bool {
 	return egress.Spec.EgressIP != "" && egress.Spec.ExternalIPPool != ""
 }
+
+// compareEgressStatus compares two Egress Statuses, ONLY taking into account IPAssigned condition
+// and NOT taking into account LastTransitionTime, returns if different
+func compareEgressStatus(currentStatus, desiredStatus crdv1a2.EgressStatus) bool {
+	if currentStatus.EgressIP != desiredStatus.EgressIP || currentStatus.EgressNode != desiredStatus.EgressNode {
+		return true
+	}
+	if len(desiredStatus.Conditions) == 0 {
+		return true
+	}
+	exists := false
+	for _, v := range currentStatus.Conditions {
+		if v.Type == crdv1a2.IPAssigned {
+			exists = true
+			if len(desiredStatus.Conditions) != 0 && (v.Message != desiredStatus.Conditions[0].Message || v.Reason != desiredStatus.Conditions[0].Reason || v.Status != desiredStatus.Conditions[0].Status) {
+				return true
+			}
+			break
+		}
+	}
+	if !exists && len(desiredStatus.Conditions) != 0 {
+		return true
+	}
+	return false
+}
```

Comment:
Could you add some unit tests for `updateEgressStatus` and `compareEgressStatus`?